### PR TITLE
test: Treat `bitcoin-wallet` binary in the same way as others

### PR DESCRIPTION
**Pull request by hebasto**
This PR makes the `bitcoin-wallet` binary path customizable in the same way how it can be done now with other ones, including `bitcoind`, `bitcoin-cli` and `bitcoin-util`.

### CODE REVIEW

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -252,9 +252,15 @@ def setup(self):
...
         )
+        fname_bitcoinwallet = os.path.join(
+            config["environment"]["BUILDDIR"],
+            "src",
+            "bitcoin-wallet" + config["environment"]["EXEEXT"],
+        )
         self.options.bitcoind = os.getenv("BITCOIND", default=fname_bitcoind)
         self.options.bitcoincli = os.getenv("BITCOINCLI", default=fname_bitcoincli)
         self.options.bitcoinutil = os.getenv("BITCOINUTIL", default=fname_bitcoinutil)
+        self.options.bitcoinwallet = os.getenv("BITCOINWALLET", default=fname_bitcoinwallet)
```
[the remark this answers is not in the document]
Comment (hebasto) [reply test/functional/test_framework/test_framework.py]: > What about making this a function to reduce code bloat while touching this?

Done.

**Review by maflcko — APPROVED** [review]
lgtm

**Review by stickies-v — APPROVED** [review]
ACK ef01337bbb3d5269db6a26e42c7bbd238000ce2f

Nice little refactoring (modulo being able to now set `BITCOINWALLET` which is new behaviour), less code duplication and functions with smaller scope.

**Comment by hebasto** [comment]
Updated ef01337bbb3d5269db6a26e42c7bbd238000ce2f -> de8b7ab2e023e28168e870629c2423dab589085e ([pr27554.03](https://github.com/hebasto/bitcoin/commits/pr27554.03) -> [pr27554.04](https://github.com/hebasto/bitcoin/commits/pr27554.04), [diff](https://github.com/hebasto/bitcoin/compare/pr27554.03..pr27554.04)):

- addressed @stickies-v's comments

File: test/functional/test_framework/test_framework.py [hunk 2]
```diff
@@ -228,6 +228,22 @@ def parse_args(self):
 
         PortSeed.n = self.options.port_seed
 
+    def set_binary_paths(self):
```
[the remark this answers is not in the document]
Comment (stickies-v) [reply test/functional/test_framework/test_framework.py:232]: Oops sorry, the docstring is meant to go underneath the function `def`, not above. My bad.

[the remark this answers is not in the document]
Comment (hebasto) [reply test/functional/test_framework/test_framework.py:231]: Adjusted.

**Review by stickies-v — APPROVED** [review]
re-ACK f6d7636be4eb0b19878428906dd5e394df7d07a2

**Review by maflcko — APPROVED** [review]

**Pull request merged**